### PR TITLE
[Selenium] Create method for performing specified functionality with ignoring provided exception

### PR DESCRIPTION
**Pull request by Ohrimenko1988**
### What does this PR do?
Create method for performing specified functionality with ignoring provided exception

For example: 
```
seleniumWebDriverHelper.waitPerformWithExceptionIgnoring(
()-> seleniumWebDriverHelper.waitAndClick(By.xpath("someXpath")),
StaleElementReferenceException.class)
```
method "waitAndClick" will be performed until success or timeout end. Besides if during performing the specified method will cause "StaleElementReferenceException" the method will be rerun, exception will be ignored and element will be found again.

### What issues does this PR fix or reference?
Issue: https://github.com/eclipse/che/issues/11241

### CODE REVIEW

**Review by musienko-maxim — APPROVED** [review]

**Review by SkorikSergey — APPROVED** [review]

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/WebDriverWaitFactory.java [hunk 1]
```diff
@@ -43,6 +44,14 @@ public WebDriverWait get(int timeoutInSec) {
     return new WebDriverWait(seleniumWebDriver, timeoutInSec).withMessage(messageSupplier);
   }
 
+  public FluentWait<WebDriver> get(
+      int timeoutInSec, Class<? extends Throwable> ignoredExceptionType) {
+    return new FluentWait<WebDriver>(seleniumWebDriver)
+        .withTimeout(timeoutInSec, TimeUnit.SECONDS)
+        .pollingEvery(200, TimeUnit.MILLISECONDS)
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/WebDriverWaitFactory.java]: `.pollingEvery(200, TimeUnit.MILLISECONDS)` looks redundant so as default delay between the attempts is preset to 500 millisecs inside FluentWait itself.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/WebDriverWaitFactory.java [hunk 2]
```diff
@@ -43,6 +44,14 @@ public WebDriverWait get(int timeoutInSec) {
     return new WebDriverWait(seleniumWebDriver, timeoutInSec).withMessage(messageSupplier);
   }
 
+  public FluentWait<WebDriver> get(
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/WebDriverWaitFactory.java]: JavaDocs is needed.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java [hunk 3]
```diff
@@ -69,6 +69,10 @@ public SeleniumWebDriverHelper(
     this.uploadUtil = uploadFileUtil;
   }
 
+  public interface VoidSupplier {
```
Comment (dmytro-ndp) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/webdriver/SeleniumWebDriverHelper.java]: We can get rid of `VoidSupplier` by using `Supplier<Void>`.

**Comment by dmytro-ndp** [comment]
@Ohrimenko1988: please, take a look at small improvements here https://github.com/eclipse/che/pull/11246/commits/239ec30be60a2a1634943898b93251a183d90f59
Example of usage:
```
    seleniumWebDriverHelper.waitNoExceptions(
        new Supplier<Void>() {
          private int counter = 3;

          @Override
          public Void get() {
            if (counter-- > 0) {
              LoggerFactory.getLogger(DialogAboutTest.class).info("exception " + counter);
              WaitUtils.sleepQuietly(2);
              throw new StaleElementReferenceException("exception " + counter);
            }

            LoggerFactory.getLogger(DialogAboutTest.class).info("return null");
            return null;
          }
        },
        StaleElementReferenceException.class,
        20);
```

**Review by artaleks9 — APPROVED** [review]

**Comment by tsmaeder** [comment]
To be honest, I don't think this is a good approach to make the ci tests more stable in the long run: we just paper over the fact that we don't understand why the StaleElementExceptions are happening. When wo do stuff the changes the elements in the page, we should wait after the change until the system has settled again. Not just continue and then wait the next time we want to work with the dom again.

**Comment by Ohrimenko1988** [comment]
@tsmaeder The way which you described is only one approach, we can do by another way (like in this PR). And it will be used only in some places if compare with existing waits in this places, this PR approach works even faster.

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/826/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/826//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11246
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**